ligo/types.mligo

```diff
@@ -300,18 +300,30 @@ type y_to_x_rec_param = x_to_y_rec_param
 
 type result = (operation list) * storage
 
+type x_to_x_prime_param = {
+    (* X tokens to sell. *)
+    dx : nat ;
+    (* Address of another segmented-cfmm contract. *)
+    x_prime_contract : address ;
+    (* The transaction won't be executed past this point. *)
+    deadline : timestamp ;
+    (* The transaction won't be executed if buying less than the given amount of X' tokens. *)
+    min_dx_prime : nat ;
+    (* Recipient of dx'. *)
+    to_dx_prime : address ;
+}
 
 (* Entrypoints *)
 
 type views =
   | IC_sum of int
 
 type parameter =
-  | X_to_Y of x_to_y_param
-  | Y_to_X of y_to_x_param
+  | X_to_y of x_to_y_param
+  | Y_to_x of y_to_x_param
+  | X_to_x_prime of x_to_x_prime_param (* equivalent to token_to_token *)
```

Comment:
On another note, I also rename the entrypoints to be all lower-cased. Let me know if this is not preferred.

Reply:
Yep, this sounds correct to me, as it matches the naming used in the spec :ok_hand: